### PR TITLE
feat: AnalysisSuggestions Livewire component (#167)

### LINKED ISSUE #167: Suggestion UI: analysis results banner on Connect Bank page
## Description

Create a Livewire component that displays pipeline analysis suggestions on the Connect Bank page. Users can accept or reject each suggestion, which commits or discards the detected data.

**Parent Epic:** #160
**Depends on:** #161 (Foundation), #163 (Recurring Stage), #164 (Primary Account Stage), #165 (Pay Cycle Stage)

## Livewire Component

**File:** `app/Livewire/AnalysisSuggestions.php`
**View:** `resources/views/livewire/analysis-suggestions.blade.php`

Embedded on the Connect Bank page. Queries pending suggestions grouped by type.

### Display Sections (ordered)

1. **Primary Account** — "We detected [account name] as your primary account" — Accept/Reject buttons
2. **Pay Cycle** — "You appear to be paid \$X [frequency], next on [date]" — Accept/Reject with inline edit fields for amount, frequency, date
3. **Recurring Transactions** — List of detected patterns with description, amount, frequency, match count. Accept/Reject per item. Accept allows setting category before confirming.

### Acceptance Actions

**PrimaryAccount accepted:**
```php
$user->update(['primary_account_id' => $payload['account_id']]);
$suggestion->update(['status' => Accepted, 'resolved_at' => now()]);
```

**PayCycle accepted:**
```php
$user->update([
    'pay_amount' => $payload['pay_amount'],
    'pay_frequency' => $payload['pay_frequency'],
    'next_pay_date' => $payload['next_pay_date'],
]);
$suggestion->update(['status' => Accepted, 'resolved_at' => now()]);
```

**RecurringTransaction accepted (in DB transaction):**
1. Create `PlannedTransaction` from payload
2. Link matched transactions: `Transaction::whereIn('id', $payload['matched_transaction_ids'])->update(['planned_transaction_id' => $planned->id])`
3. If user set a category, apply to PlannedTransaction and all linked transactions
4. Mark suggestion as Accepted

**Rejection:**
```php
$suggestion->update(['status' => Rejected, 'resolved_at' => now()]);
```

### UI Design

- Inline card/banner style using Flux UI components
- Only visible when pending suggestions exist
- Grouped by type with clear section headers
- Accept/Reject buttons with confirmation states
- Recurring transactions shown as a scrollable list with key details (description, amount, frequency, match count)

## Tasks

- [ ] Create `AnalysisSuggestions` Livewire component
- [ ] Create Blade view with Flux UI components
- [ ] Implement `acceptSuggestion(int $id)` action with type-specific logic
- [ ] Implement `rejectSuggestion(int $id)` action
- [ ] Implement inline category selection for recurring transaction acceptance
- [ ] Embed component on Connect Bank page
- [ ] Write Livewire component tests:
  - Displays pending suggestions grouped by type
  - Accept primary account updates User model
  - Accept pay cycle updates User pay fields
  - Accept recurring creates PlannedTransaction and links transactions
  - Reject updates status and hides suggestion
  - Component hidden when no pending suggestions
- [ ] Run `op ci` to verify

## Acceptance Criteria

- Suggestions appear on Connect Bank page after sync completes
- Each suggestion type has appropriate accept/reject UI
- Accepting commits real data correctly
- Rejecting hides suggestion and prevents re-suggestion for 90 days
- Component is hidden when no pending suggestions exist
- Follows existing Flux UI patterns and Livewire conventions
- All tests pass, `op ci` green

### PR DESCRIPTION
**Pull request by robwilde**
## Summary

- Add `AnalysisSuggestions` Livewire component that displays pending pipeline suggestions (primary account, pay cycle, recurring transactions) on the Connect Bank page
- Users can accept or reject each suggestion type — accepting commits the detected data to their profile, rejecting dismisses the suggestion
- Component polls every 30s to pick up new suggestions created by async pipeline jobs
- 16 Pest feature tests covering rendering, accept/reject flows, authorization guards, validation, and edge cases

Closes #167

## Test plan

- [x] `op test.filter AnalysisSuggestions` — 16/16 tests pass
- [x] `op lint.dirty` — Pint formatting clean
- [x] `op ci` — full CI green (1186 passed, 0 failures)
- [ ] Manual: visit Connect Bank page with pending suggestions, verify accept/reject flows render and work correctly
- [ ] Manual: verify component auto-hides when no pending suggestions exist

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: app/Livewire/AnalysisSuggestions.php [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+            foreach ($recurringTransactions as $suggestion) {
+                if (! array_key_exists($suggestion->id, $this->recurringCategories)) {
+                    $this->recurringCategories[$suggestion->id] = $suggestion->payload['category_id'] ?? null;
+                }
+            }
+        }
+
+        return view('livewire.analysis-suggestions', [
+            'suggestions' => $suggestions,
+            'categories' => Category::visible()->orderBy('name')->get(),
```
[the remark this answers is not in the document]
Comment (robwilde) [reply app/Livewire/AnalysisSuggestions.php]: Good catch — `fullPath()` walks the parent chain lazily. Fixed by eager-loading `parent.parent` in the category query. Committed in the follow-up.

File: app/Livewire/AnalysisSuggestions.php [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+                'user_id' => $user->id,
+                'account_id' => $payload['account_id'],
+                'amount' => $payload['amount'],
+                'direction' => $payload['direction'],
+                'description' => $payload['clean_description'],
+                'start_date' => $payload['start_date'],
+                'frequency' => $payload['frequency'],
+                'is_active' => true,
+                'category_id' => $categoryId,
+            ]);
```
[the remark this answers is not in the document]
Comment (robwilde) [reply app/Livewire/AnalysisSuggestions.php:101]: Valid — added a guard that checks `Category::visible()->where('id', $categoryId)->exists()` before use, falling back to `null` for hidden or nonexistent categories. Also added a test covering the hidden category scenario.
